### PR TITLE
CDAP 19698 Feature - File Parsing Panel

### PR DESCRIPTION
**Pull request by adityadivami**
# TITLE
Feature - File Parsing Panel

## Description
Summary of changes
This PR includes the File parsing panel feature

## Files to be Reviewed
All files are to be reviewed, no exceptions

## PR Type
- [ ] Bug Fix
- [x] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [Jira issue #](fill-in.org)
https://cdap.atlassian.net/browse/CDAP-19698

## Test Plan
1. FE_02_001 - Verify whether the parsing popup window is displayed or not. - Pass
2. FE_02_002 - Verify whether the "Parsing" text is displayed as the title on the popup screen or not - Pass
3. FE_02_003 - Verify the "download icon" is displayed on the right side of the pop-up screen - Pass
4. FE_02_004 - Verify the "Import schema" text is displayed in between the download icon & close icon of the pop-up screen - Pass
5. FE_02_005 - Verify the "close icon" is displayed on the right end of the pop-up screen - Pass
6. FE_02_006 - Verify by clicking on the 'Close icon' and check pop screen is closing or not - Pass
7. FE_02_007 - Verify the action after clicking on the 'Close icon'. - Pass
8. FE_02_008 - Verify whether the format field dropdown list fields are displayed or not. - Pass
9. FE_02_009 - Verify whether the Encoding drop-down list fields are displaying or not. - Pass
10. FE_02_010 - Verify by changing the selection of 'the "Format & Encoding" drop-down list fields. - Pass
11. FE_02_011 - Verify the Enable Quoted Value checkbox and check the user is able to check/uncheck the box - Pass
12. FE_02_012 - Verify the Use First Row as Header checkbox and check the user is able to check/uncheck the box - Pass
13. FE_02_013 - Verify after selecting the Use First Row as Header checkbox and verify the data interchanging. - Pass
14. FE_02_014 - Verify the info icon beside once applied parsing settings cannot be changed. displayed or not. - Pass
15. FE_02_015 - Verify the 'Apply' Text button by mouse hovering - Pass
16. FE_02_016 - Verify by clicking on the 'Apply' button  - Pass
17. FE_02_017 - Verify the redirection after clicking on the 'Apply' button. - Pass
18. FE_02_018 - Verify after the 'Apply' button whether the popup message is displayed with a valid/successful submission message or not. - Pass
19. FE_02_019 - Verify after the 'Apply' button whether the popup message is displayed with an invalid/warning message or not. - Pass

## Screenshots
Screenshot 2022-11-16 at 12 12 32 AM

Screenshot 2022-11-16 at 12 12 38 AM

Screenshot 2022-11-16 at 12 12 50 AM

Screenshot 2022-11-16 at 12 13 01 AM

### CODE REVIEW

**Review by seanfeld — CHANGES_REQUESTED** [review]
Please start adding which files need to be skipped - also in the screenshot it looks like the incorrect font is used in the dropdowns

**Comment by adityadivami** [comment]
> Please start adding which files need to be skipped - also in the screenshot it looks like the incorrect font is used in the dropdowns

All files are to be reviewed as part of this PR

File: app/cdap/components/GridTable/index.tsx [hunk 1]
```diff
@@ -57,6 +58,32 @@ export default function GridTable() {
     },
   ]);
 
+  const [isFirstWrangle, setIsFirstWrangle] = useState<boolean>(false);
```
Comment (njbriggs) [inline app/cdap/components/GridTable/index.tsx:61]: Please add some comments about what `isFirstWrangle` means

File: app/cdap/components/ParsingDrawer/Components/ParsingPopupBody/index.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+          defaultValue={CHAR_ENCODING_OPTIONS[0].value}
+          fullWidth
+          value={selectedEncodingValue[0]?.value}
+          onChange={(event: ChangeEvent<HTMLSelectElement>) =>
+            changeEventListener(event.target.value, 'fileEncoding')
+          }
+          options={CHAR_ENCODING_OPTIONS}
+        />
+      </FormFieldWrapper>
+      <CheckBox
```
Comment (njbriggs) [inline app/cdap/components/ParsingDrawer/Components/ParsingPopupBody/index.tsx:123]: These fields should not be hardcoded. Please look at how the current parsing dialog handles different file types.

File: app/cdap/components/ParsingDrawer/Components/ParsingPopupBody/parsingOptions.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import T from 'i18n-react';
+
+const PREFIX = 'features.WranglerNewUI.ParsingDrawer.SetCharEncoding';
+const SUFFIX = 'features.WranglerNewUI.ParsingDrawer.Parse';
+
+export const CHAR_ENCODING_OPTIONS = [
```
Comment (njbriggs) [inline app/cdap/components/ParsingDrawer/Components/ParsingPopupBody/parsingOptions.ts:22]: These are provided by the API and should not be hardcoded.

**Pull request closed without merging**